src/unstract/sdk/tool/entrypoint.py

```diff
@@ -1,11 +1,22 @@
+import logging
+import signal
+
 from unstract.sdk.tool.base import BaseTool
 from unstract.sdk.tool.executor import ToolExecutor
 from unstract.sdk.tool.parser import ToolArgsParser
 
+logger = logging.getLogger(__name__)
 
 class ToolEntrypoint:
     """Class that contains methods for the entrypoint for a tool."""
 
+    @staticmethod
+    def _signal_handler(signum: int, frame: Any) -> None:
+        """Handle SIGTERM and SIGINT signals."""
+        signal_name = "SIGTERM" if signum == signal.SIGTERM else "SIGINT"
```

Comment:
@kirtimanmishrazipstack I think it will be better to get the signal_name like below. So that it will be future proof. Also accidentally if we are trapping some other signals also this will make sure we are logging correctly
```suggestion
       sig = signal.Signals(signum)
       signal_name = sig.name
```